sprintvotingapp/sprint/views.py

```diff
@@ -0,0 +1,536 @@
+
+
+                    elif vote:
+                        raise AlreadyPresentException
+
+                    serializer = VoteSerializer(data=parameter)
+                    serializer.is_valid(raise_exception=True)
+                    serializer.save()
+
+            return Response(
+                {
+                    "message": "Voting Successful"
+                }, status=status.HTTP_201_CREATED
+            )
+        except InsertionError:
+            return Response(
+                {
+                    "message": "You cannot Vote for yourself",
+                    "data": parameter
+                }, status=status.HTTP_400_BAD_REQUEST
+            )
+
+        except AlreadyPresentException:
+            return Response(
+                {
+                    "message": "Parameter addition Unsuccessful-already voted on this parameter",
+                    "data": parameter
+                }, status=status.HTTP_400_BAD_REQUEST
+            )
+        except Exception as e:
+            logging.error(e)
+
+            return Response(
+                {
+                    "message": "Exception Occurred",
+                    "error": str(e)
+                }, status=status.HTTP_400_BAD_REQUEST
+            )
+
+    @verify_token
+    def get(self, request, id):
+        """
+        For getting the details of vote done by a user
+        :param request:
+        :param id: User id of how is login
+        :return: Response
+        """
+        try:
+
+            votes = Votes.objects.filter(vote_by=request.data.get("user_id"), sprint_id=id)
+            serializer = VoteSerializer(votes, many=True)
+            vote_list = list()
+            if serializer.data:
+                for vote in serializer.data:
+                    vote_list.append({"parameter_id": vote.get("parameter_id"), "vote_to": vote.get("vote_to")})
+                vote_data = {
+                    "Vote_by": request.data.get("user_id"),
+                    "sprint_id": id,
+                    "vote_details": vote_list
+                }
+                return Response(
+                    {
+                        "message": "Here is your data",
+                        "data": vote_data
+                    }, status=status.HTTP_200_OK
+                )
+            return Response(
+                {
+                    "message": "He has Not vote for any parameter",
+
+                }, status=status.HTTP_404_NOT_FOUND
+            )
+        except ValidationError:
+            return Response(
+                {
+                    "message": "validation error"
+                }, status=status.HTTP_400_BAD_REQUEST
+            )
+        except Exception as e:
+            logging.error(e)
+            return Response(
+                {
+                    "message": "Error Occurred",
+                    "error": str(e)
+                }, status=status.HTTP_500_INTERNAL_SERVER_ERROR
+            )
+
+    @verify_token
```

Comment:
should be an atomic transaction? Either all votes should be updated or none. We shouldn't allow partial updates of a few votes and then throw an error.

Reply:
I am using @transaction_atomic ,if any exception got catch then it rollback the changes